graphene_sqlalchemy/types.py

```diff
@@ -345,3 +362,105 @@ def enum_for_field(cls, field_name):
+
+    Usage:
+
+        class MyModel(Base):
+            id = Column(Integer(), primary_key=True)
+            name = Column(String())
+
+        class MyType(SQLAlchemyObjectType):
+            class Meta:
+                model = MyModel
+    """
+
+    @classmethod
+    def __init_subclass_with_meta__(cls, _meta=None, **options):
+        if not _meta:
+            _meta = SQLAlchemyObjectTypeOptions(cls)
+
+        super(SQLAlchemyObjectType, cls).__init_subclass_with_meta__(
+            _meta=_meta, **options
+        )
+
+
+class SQLAlchemyInterfaceOptions(InterfaceOptions):
+    model = None  # type: sqlalchemy.Model
+    registry = None  # type: sqlalchemy.Registry
+    connection = None  # type: sqlalchemy.Type[sqlalchemy.Connection]
+    id = None  # type: str
+
+
+class SQLAlchemyInterface(SQLAlchemyBase, Interface):
+    """
+    This type represents the GraphQL Interface. It reflects on the
+    given SQLAlchemy model, and automatically generates an Interface
+    using the column and relationship information defined there. This
+    is used to construct interface relationships based on polymorphic
+    inheritance hierarchies in SQLAlchemy.
+
+    Please note that by default, the "polymorphic_on" column is *not*
+    generated as a field on types that use polymorphic inheritance, as
+    this is considered an implentation detail. The idiomatic way to
+    retrieve the concrete GraphQL type of an object is to query for the
+    `__typename` field.
+
+    Usage (using joined table inheritance):
+
+        class MyBaseModel(Base):
+            id = Column(Integer(), primary_key=True)
+            type = Column(String())
+            name = Column(String())
+
+        __mapper_args__ = {
+            "polymorphic_on": type,
+        }
+
+        class MyChildModel(Base):
+            date = Column(Date())
+
+        __mapper_args__ = {
+            "polymorphic_identity": "child",
+        }
+
+        class MyBaseType(SQLAlchemyInterface):
+            class Meta:
+                model = MyBaseModel
+
+        class MyChildType(SQLAlchemyObjectType):
+            class Meta:
+                model = MyChildModel
+                interfaces = (MyBaseType,)
+    """
+
+    @classmethod
+    def __init_subclass_with_meta__(cls, _meta=None, **options):
+        if not _meta:
+            _meta = SQLAlchemyInterfaceOptions(cls)
+
+        super(SQLAlchemyInterface, cls).__init_subclass_with_meta__(
+            _meta=_meta, **options
+        )
+
+        # make sure that the model doesn't have a polymorphic_identity defined
+        if hasattr(_meta.model, "__mapper__"):
+            polymorphic_identity = _meta.model.__mapper__.polymorphic_identity
+            assert (
+                polymorphic_identity is None
+            ), '{}: An interface cannot map to a concrete type (polymorphic_identity is "{}")'.format(
+                cls.__name__, polymorphic_identity
+            )
```

Comment:
@polgfred I just realized that this might have been a mistake to add. Sometimes we have a base type that is extended by other types but has a polymorphic identity. In this case, there would be an object type implementing the interface with no additional fields. We might need to revert hat bit, what do you think?